---
float64 x
float64 y
string direction
float64 battery
